README.md
# google-cloud-speech-api
Testing the Google Cloud Speech API


history.txt
To do
1. Setup (one time only)

#set variable for file name

#
export GOOGLE_APPLICATION_CREDENTIALS=gc-speech-api-79f84132db5f.json

# 
TOKEN=$(gcloud auth application-default print-access-token)

# This edits the metadata the api expects.  make sure it’s the same as the file you are converting.
cat > Lewis.json <<EOF
{
  "config": {
      "encoding":"FLAC",
      "sampleRateHertz": 44100,
      "languageCode": "en-US"
  },
  "audio": {
      "uri":"gs://audio-test-123/Lewis.flac"
  }
}
EOF

#
curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:longrunningrecognize     -d @Lewis.json

#
echo <<EOF
curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/6152850171522197711 > Lewis-output.json
EOF




    1  cat > sync-request.json
    2  gcloud iam service-accounts create my-sa-123 --display-name "my service account"
    3  gcloud auth activate-service-account --key-file=service-account-key-file
    4  ls-la
    5  ls -la
    6  gcloud auth activate-service-account --key-file=gc-speech-api-79f84132db5f.json
    7  gcloud auth application-default print-access-token
    8  access_token
    9    curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer access_token"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   10  gcloud auth application-default print-access-token
   11    curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer ya29.GpsBlQS7XA3DrdlSDS9mlbJm8s-SFr5cd8Cv34YP3cD_OtshEkW7ajDRUHBigTed0P5pnCrLHF38rXbnWwt0_gG2Z-8EcxurMKxp0wQamead-0PcwE9cUsQ88oD2YBzcO://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   12  OyIP2SDlXKK4I0HO2M0I1ec3FPjpShwJTp6Hgo9zJnkgNfEp0p4BxmBLo_HuoexO_Z31Tpvoj4vxS9qzh0
   13  TOKEN=$(gcloud auth application-default print-access-token)
   14  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   15  export GOOGLE_APPLICATION_CREDENTIALS=key.json
   16  ls -la
   17  export GOOGLE_APPLICATION_CREDENTIALS=gc-speech-api-79f84132db5f.json
   18  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   19  TOKEN=$(gcloud auth application-default print-access-token)
   20  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   21  cat sync-request.json 
   22  cp sync-request.json Lewis.json
   23  nano Lewis.json
   24  cat Lewis.json
   25  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @Lewis.json
   26  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   27  cat sync-request.json 
   28  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @Lewis.json
   29  cat Lewis.json
   30  nano Lewis.json
   31  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @Lewis.json
   32  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:longrunningrecognize     -d @Lewis.json
   33  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/opperations/6152850171522197711
   34  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/6152850171522197711
   35  curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/615285017152219771
   36  curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/6152850171522197711
   37  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   38  TOKEN=$(gcloud auth application-default print-access-token)
   39  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   40  export GOOGLE_APPLICATION_CREDENTIALS=gc-speech-api-79f84132db5f.json
   41  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   42  TOKEN=$(gcloud auth application-default print-access-token)
   43  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:recognize     -d @sync-request.json
   44  curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:longrunningrecognize     -d @sync-request.json
   45  curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/6152850171522197711
   46  curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/1077066012408181686
   47  curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/6152850171522197711

simple-test.sh
#!/bin/bash

## by hand run this command with whatever your key is called
## (replace foo-key.json with the name of your key):
##
##    cp $HOME/foo-key.json $HOME/simple-test-key.json

# points to the service account key file
export GOOGLE_APPLICATION_CREDENTIALS=$HOME/simple-test-key.json

# gets a new access token based on the service account key file
TOKEN=$(gcloud auth application-default print-access-token)

# This edits the metadata the API expects.  make sure it’s the same as the file you are converting.
cat > simple-test.json <<EOF
{
  "config": {
      "encoding":"FLAC",
      "sampleRateHertz": 16000,
      "languageCode": "en-US"
  },
  "audio": {
      "uri":"gs://cloud-samples-tests/speech/brooklyn.flac"
  }
}
EOF

#
curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:longrunningrecognize     -d @simple-test.json > simple-test.status
JOBID=$(cat simple-test.status | jq -r .name)
echo Waiting for job $JOBID to finish
sleep 3

curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/$JOBID > simple-test-output.json
cat simple-test-output.json

cat <<EOF
Run these commands again if the job is not yet done:

curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/$JOBID > simple-test-output.json
cat simple-test-output.json
EOF


speech-api.sh
#!/bin/bash

NAME=${1:-brooklyn}

# To do
# 1. Setup (one time only) 

# enable the APIs: speech, compute engine, cloud storage

# create a service account 

# download the service account key file to your laptop, then upload it to google cloud shell

# set variable for file name

# make sure for the default test to download this file and upload it to your own bucket:
#  gsutil cp gs://cloud-samples-tests/speech/brooklyn.flac .

## by hand run this command with whatever your key is called
## (replace foo-key.json with the name of your key):
##
##    cp $HOME/foo-key.json $HOME/simple-test-key.json

# points to the service account key file
export GOOGLE_APPLICATION_CREDENTIALS=$HOME/simple-test-key.json

# gets a new access token based on the service account key file
TOKEN=$(gcloud auth application-default print-access-token)

# This edits the metadata the API expects.  make sure it’s the same as the file you are converting.

# Add this back after encoding if you want to do it explicitly 
# "sampleRateHertz": 44100,

cat > ${NAME}.json <<EOF
{
  "config": {
      "encoding":"FLAC",
      "languageCode": "en-US"
  },
  "audio": {
      "uri":"gs://audio-test-123/${NAME}.flac"
  }
}
EOF

#
curl -s -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/speech:longrunningrecognize     -d @${NAME}.json > ${NAME}.status
JOBID=$(cat ${NAME}.status | jq -r .name)
echo Waiting for job $JOBID to finish
sleep 3

curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/$JOBID > ${NAME}-output.json
cat ${NAME}-output.json

cat <<EOF
Run these commands again if the job is not yet done:

curl -s -k -H "Content-Type: application/json"     -H "Authorization: Bearer $TOKEN"     https://speech.googleapis.com/v1/operations/$JOBID > ${NAME}-output.json
cat ${NAME}-output.json
EOF
